packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1114,6 +1115,12 @@ async function renderToHTMLOrFlightImpl(
       metadata.fetchTags = response.collectedTags.join(',')
     }
 
+    // Let the client router know how long to keep the cached entry around.
+    const staleHeader = String(response.collectedStale)
+    res.setHeader(NEXT_ROUTER_STALE_TIME_HEADER, staleHeader)
+    metadata.headers ??= {}
+    metadata.headers[NEXT_ROUTER_STALE_TIME_HEADER] = staleHeader
```

Comment:
Is this need to be manually added instead of the handling here? https://github.com/vercel/next.js/blob/canary/packages/next/src/server/app-render/app-render.tsx#L1900

Reply:
Mainly I wasn't sure whether it's better to add another metadata value or handle it inside the prerender since it's unique to prerenders.